pkgs/by-name/ko/koboldcpp/package.nix

```diff
@@ -0,0 +1,115 @@
+{ lib
+, fetchFromGitHub
+, stdenv
+, autoPatchelfHook
+, makeWrapper
+, python311Packages
+, tk
+
+, koboldLiteSupport ? true
+
+, config
+, cudaPackages ? { }
+
+, openblasSupport ? !stdenv.isDarwin
+, openblas
+
+, cublasSupport ? config.cudaSupport
+
+, clblastSupport ? stdenv.isLinux
+, clblast
+, ocl-icd
+
+, vulkanSupport ? true
+, vulkan-loader
+
+, metalSupport ? stdenv.isDarwin && stdenv.isAarch64
+
+, march ? ""
+, mtune ? ""
```

Comment:
Shouldn't we leave a comment here like this one:

```
# You can find list of x86_64 options here: https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html
# For ARM here: https://gcc.gnu.org/onlinedocs/gcc/ARM-Options.html
# If you set "march" to "native", specify "mtune" as well; otherwise, it will be set to "generic". (credit to: https://lemire.me/blog/2018/07/25/it-is-more-complicated-than-i-thought-mtune-march-in-gcc/)
# For example, if you have an AMD Ryzen CPU, you will set "march" to "x86-64" and "mtune" to "znver2"
```

Showing what options are possible?